server/scsynth/SC_PortAudio.cpp

```diff
@@ -311,11 +312,28 @@ bool SC_PortAudioDriver::DriverSetup(int* outNumSamples, double* outSampleRate)
     mDeviceInOut[0] = paNoDevice;
     mDeviceInOut[1] = paNoDevice;
     if (mWorld->hw->mInDeviceName)
-        GetPaDeviceFromName(mWorld->hw->mInDeviceName, mDeviceInOut);
+        GetPaDeviceFromName(mWorld->hw->mInDeviceName, &mDeviceInOut[0], IOType::Input);
+    if (mWorld->hw->mOutDeviceName)
+        GetPaDeviceFromName(mWorld->hw->mOutDeviceName, &mDeviceInOut[1], IOType::Output);
+
+#ifdef _WIN32
+    // if one device is specified, let's try to open another one on matching api
+    if (mDeviceInOut[0] == paNoDevice && mDeviceInOut[1] != paNoDevice)
+        mDeviceInOut[0] = Pa_GetHostApiInfo(Pa_GetDeviceInfo(mDeviceInOut[1])->hostApi)->defaultInputDevice;
+    if (mDeviceInOut[0] != paNoDevice && mDeviceInOut[1] == paNoDevice)
+        mDeviceInOut[1] = Pa_GetHostApiInfo(Pa_GetDeviceInfo(mDeviceInOut[0])->hostApi)->defaultOutputDevice;
+    // in case that didn't work, use the default device (likely through MME api)
+    if (Pa_GetDeviceInfo(mDeviceInOut[0])->hostApi != Pa_GetDeviceInfo(mDeviceInOut[1])->hostApi
```

Comment:
Pa_GetDeviceInfo may return nullptr here, making the dereference unsafe

Reply:
Thank you! I just realized that. I'll upload a fix soon.